src/lib/utils.py

```diff
@@ -108,13 +114,158 @@ def grouped_transform(
     return data
 
 
-def grouped_diff(data: DataFrame, keys: List[str], skip: List[str] = None) -> DataFrame:
+def grouped_diff(
+    data: DataFrame,
+    keys: List[str],
+    skip: List[str] = None,
+    prefix: Tuple[str, str] = ("new_", "total_"),
+    compat: bool = True,
+) -> DataFrame:
     return grouped_transform(
-        data, keys, lambda x: x.ffill().diff(), skip=skip, prefix=("new_", "total_")
+        data, keys, lambda x: x.ffill().diff(), skip=skip, prefix=prefix, compat=compat
     )
 
 
-def grouped_cumsum(data: DataFrame, keys: List[str], skip: List[str] = None) -> DataFrame:
+def grouped_cumsum(
+    data: DataFrame,
+    keys: List[str],
+    skip: List[str] = None,
+    prefix: Tuple[str, str] = ("total_", "new_"),
+    compat: bool = True,
+) -> DataFrame:
     return grouped_transform(
-        data, keys, lambda x: x.fillna(0).cumsum(), skip=skip, prefix=("total_", "new_")
+        data, keys, lambda x: x.fillna(0).cumsum(), skip=skip, prefix=prefix, compat=compat
+    )
+
+
+def stack_table(
+    data: DataFrame, index_columns: List[str], value_columns: List[str], stack_columns: List[str]
+) -> DataFrame:
+    """
+    Pivots a DataFrame's columns and aggregates the result as new columns with suffix. E.g.:
+
+    data:
+
+    idx piv val
+     0   A   1
+     0   B   2
+     1   A   3
+     1   B   4
+
+    stack_table(data, index_columns=[idx], value_columns=[val], stack_columns=[piv]):
+
+    idx val val_A val_B
+     0   3    1     2
+     1   7    3     4
+    """
+    output = data.drop(columns=stack_columns).groupby(index_columns).sum()
+
+    # Stash columns which are not part of the columns being indexed, aggregated or stacked
+    used_columns = index_columns + value_columns + stack_columns
+    stash_columns = [col for col in data.columns if col not in used_columns]
```

Comment:
Yes, this is a good point. I want to take a second look at `grouped_transform` after I remove the `grouped_diff` and `grouped_cumsum` methods from all scripts and we can simplify/refactor, since they will become methods private to the module